src/ConfigAggregator.php

```diff
@@ -92,6 +96,34 @@ public function getMergedConfig()
         return $this->config;
     }
 
+    /**
+     * Throw an exception if duplicate config providers are found.
+     *
+     * @param ProviderIterable $providers
+     * @return ProviderIterable
+     * @throws InvalidConfigProviderException
+     */
+    private function validateNoDuplicateProviders(iterable $providers): iterable
+    {
+        $uniqueProviders = [];
+
+        foreach ($providers as $provider) {
+            if (! is_string($provider) && ! is_object($provider)) {
+                continue;
+            }
+
+            if (in_array($provider, $uniqueProviders, true)) {
+                throw InvalidConfigProviderException::fromDuplicateProvider(
+                    is_string($provider) ? $provider : get_debug_type($provider)
```

Comment:
Not 100% sure but afair `get_debug_type` keeps string if string input is given. Maybe double check that, but I remember something. 🤔

Reply:
If I use `get_debug_type($provider)` on strings as well, then the error thrown would look something like:

> string is already registered...

which does not provide much info on what went wrong.

With my current implementation it will throw:

> Some\Provider is already registered...

which is much more relevant.

Reply:
Ah okay, I was expecting that it wont change the string as it is already a string. good to know, thanks for clarification. havent had time to play around with that function and since I mostly write via my phone, I did not tested it 👍🏻